plugins/ingest/src/main/java/org/elasticsearch/plugin/ingest/transport/simulate/SimulatedItemResponse.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.plugin.ingest.transport.simulate;
+
+import org.elasticsearch.ExceptionsHelper;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Streamable;
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentBuilderString;
+import org.elasticsearch.ingest.Data;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+public class SimulatedItemResponse implements Streamable, ToXContent {
+
+    private Data data;
+    private List<ProcessorResult> processorResultList;
+    private Throwable failure;
+
+    public SimulatedItemResponse() {
+
+    }
+
+    public SimulatedItemResponse(Data data) {
+        this.data = data;
+    }
+
+    public SimulatedItemResponse(List<ProcessorResult> processorResultList) {
+        this.processorResultList = processorResultList;
+    }
+
+    public SimulatedItemResponse(Throwable failure) {
+        this.failure = failure;
+    }
+
+    public boolean isFailed() {
```

Comment:
I thought it would be consistent with an document level response that is received when `verbose=false`. I am on the fence about this. Since we continue to run other processors even if there is a failure, I will remove this item level flag